contracts/SushiMaker.sol

```diff
@@ -1,264 +1,112 @@
-// SPDX-License-Identifier: MIT
+// SPDX-License-Identifier: GPLV3
 
 // P1 - P3: OK
 pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
 import "./libraries/SafeMath.sol";
-import "./libraries/SafeERC20.sol";
 
 import "./uniswapv2/interfaces/IUniswapV2ERC20.sol";
 import "./uniswapv2/interfaces/IUniswapV2Pair.sol";
 import "./uniswapv2/interfaces/IUniswapV2Factory.sol";
+import './uniswapv2/libraries/UniswapV2Library.sol';
+import './uniswapv2/libraries/TransferHelper.sol';
+import "@boringcrypto/boring-solidity/contracts/BoringOwnable.sol";
+import "@boringcrypto/boring-solidity/contracts/BoringBatchable.sol";
+import "@boringcrypto/boring-solidity/contracts/libraries/BoringERC20.sol";
 
-import "./Ownable.sol";
 
 // SushiMaker is MasterChef's left hand and kinda a wizard. He can cook up Sushi from pretty much anything!
 // This contract handles "serving up" rewards for xSushi holders by trading tokens collected from fees for Sushi.
 
 // T1 - T4: OK
-contract SushiMaker is Ownable {
+contract SushiMaker is BoringOwnable, BoringBatchable {
     using SafeMath for uint256;
-    using SafeERC20 for IERC20;
+    using BoringERC20 for IERC20;
 
     // V1 - V5: OK
-    IUniswapV2Factory public immutable factory;
+    address public immutable factory;
```

Comment:
This is not needed anymore.

Reply:
It is still needed to specify the factory to get the actual pairs, either as immutable or constant